File: mario.c
// CS50x 2019 Problem Set 1 - mario.c (more)
// 2019 Aug 22

#include <cs50.h>
#include <stdio.h>

int main(void)
{
    //prompt for height
    int height;
    do
    {
        height = get_int("Height: ");
    }
    while (height < 1 || height > 8);

    //printing #
    for (int i = 0; i < height; i++)
    {
        for (int j = 0; j < height; j++)
        {
            if (j < (height - i - 1))
            {
                printf(" ");
            }
            else
            {
                printf("#");
            }
        }
        printf("  ");
        for (int k = 0; k < i + 1; k++)
        {
            printf("#");
        }
        printf("\n");
    }
}

File: mario.py
# /more/mario.py
# Oct 3, 2019

from cs50 import get_int

# prompt user for positive number
while True:
    height = get_int("Height: ")
    if height > 0 and height < 9:
        break

# print out the column and rows
for i in range(height):
    for j in range(height):
        # print the space follow by #
        if j < (height - i - 1):
            print(" ", end='')
        else:
            print("#", end='')
        # print return to prompt
    print("  ", end='')

    # print the other half of row
    for k in range(i + 1):
        print("#", end='')
    print("")

# $ python mario.py
# Height: -5
# Height: five
# Height: 24
# Height: 4
#    #  #
#   ##  ##
#  ###  ###
# ####  ####
